### PR TITLE
refactor: add optionType for Radio internally

### PR DESCRIPTION
**Pull request by vagusX**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/pull/34740

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Add `optionType` property to Radio component for internal usage      |
| 🇨🇳 Chinese |   给 Radio 组件增加  `optionType` 属性供内部使用        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap [hunk 1]
```diff
@@ -19156,18 +19156,18 @@ exports[`ConfigProvider components Radio prefixCls 1`] = `
...
       >
         <input
           checked=""
-          class="prefix-Radio-input"
+          class="prefix-Radio-button-input"
           type="radio"
         />
         <span
-          class="prefix-Radio-inner"
+          class="prefix-Radio-button-inner"
```
Comment (vagusX) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap:19170]: image

**Comment by zombieJ** [comment]
ConfigProvider 透传错了 prefixCls ，感觉会 break。弄到 feature branch 做 fix 如何？

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap (hunk 1 above)
Comment (vagusX) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap:19170]: 这是线上的 html，这样看起来 snapshot 就是错的诶

**Review by zombieJ — APPROVED** [review]

File: components/radio/radioButton.tsx [hunk 2]
```diff
@@ -12,12 +12,12 @@ const RadioButton = (props: RadioButtonProps, ref: React.Ref<any>) => {
...
 
   const { prefixCls: customizePrefixCls, ...radioProps } = props;
-  const prefixCls = getPrefixCls('radio-button', customizePrefixCls);
+  const prefixCls = getPrefixCls('radio', customizePrefixCls);
   if (radioGroupContext) {
     radioProps.checked = props.value === radioGroupContext.value;
     radioProps.disabled = props.disabled || radioGroupContext.disabled;
   }
-  return <Radio prefixCls={prefixCls} {...radioProps} type="radio" ref={ref} />;
+  return <Radio prefixCls={prefixCls} {...radioProps} type="radio" optionType="button" ref={ref} />;
```
Comment (zombieJ) [inline components/radio/radioButton.tsx:20]: 这边可以走 context 不

Comment (vagusX) [reply components/radio/radioButton.tsx:20]: 我看看

File: components/radio/radioButton.tsx [hunk 3]
```diff
@@ -12,12 +12,17 @@ const RadioButton = (props: RadioButtonProps, ref: React.Ref<any>) => {
   const { getPrefixCls } = React.useContext(ConfigContext);
 
   const { prefixCls: customizePrefixCls, ...radioProps } = props;
-  const prefixCls = getPrefixCls('radio-button', customizePrefixCls);
-  if (radioGroupContext) {
-    radioProps.checked = props.value === radioGroupContext.value;
-    radioProps.disabled = props.disabled || radioGroupContext.disabled;
+  const prefixCls = getPrefixCls('radio', customizePrefixCls);
+
+  if (radioGroupContext?.optionType) {
```
Comment (vagusX) [inline components/radio/radioButton.tsx:17]: 这个分支要不要删掉 @zombieJ

File: components/radio/radio.tsx [hunk 4]
```diff
@@ -22,17 +24,22 @@ const InternalRadio: React.ForwardRefRenderFunction<HTMLElement, RadioProps> = (
...
     props.onChange?.(e);
-    context?.onChange?.(e);
+    groupContext?.onChange?.(e);
   };
 
   const { prefixCls: customizePrefixCls, className, children, style, ...restProps } = props;
-  const prefixCls = getPrefixCls('radio', customizePrefixCls);
+  const radioPrefixCls = getPrefixCls('radio', customizePrefixCls);
+  const prefixCls =
+    (groupContext?.optionType || radioOptionTypeContext) === 'button'
```
Comment (vagusX) [inline components/radio/radio.tsx:33]: 这里的优先级是否合适? @zombieJ

File: components/radio/radioButton.tsx (hunk 3 above)
Comment (zombieJ) [reply components/radio/radioButton.tsx:17]: 感觉可以删了

**Comment by vagusX** [comment]
单测覆盖我来加一下测试

File: components/radio/__tests__/radio-button.test.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
+import React from 'react';
```
Comment (vagusX) [inline components/radio/__tests__/radio-button.test.js:1]: 把 RadioButton 的单测独立出来，补上了

**Pull request merged**